src/extension.c

```diff
@@ -149,7 +135,13 @@ extension_is_loaded(void)
 	 */
 	id = get_extension_oid(EXTENSION_NAME, true);
 
-	if (OidIsValid(id))
+	/*
+	 * the proxy table check is necessary because during an extension drop the
+	 * proxy table will be deleted before the extension itself. You don't want
+	 * to mark the extension as CREATED in this case
+	 */
+
+	if (OidIsValid(id) && proxy_table_exists())
```

Comment:
I am all for simplifying this if we feel confident that it is doing the right thing. But please document the expected events that occur and their order. I agree that we can probably trust that the proxy table is dropped before the extension drop finishes, but we cannot know at which point the proxy table is removed after one issues, e.g., `DROP EXTENSION`. Before we had a state immediately following `DROP EXTENSION` and now we just know at some point after that and before the extension is fully removed the proxy table will be removed. In the meantime we still believe the extension is loaded. Does this matter? Again, I am not too keen on relying on these incidental (implicit) order of events, not least because it is hard to understand.